### PR TITLE
Add support for HuggingFace Embedding models

### PR DESCRIPTION
**Pull request by sunilkumardash9**
The PR aims to add support for embedding models hosted on HuggingFace.

### CODE REVIEW

**Comment by ddematheu** [comment]
Thank you for your contribution. Reviewing the PR.

File: neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @property
+    def required_properties(self) -> List[str]:
+        return ["token"]
+
+    @property
+    def optional_properties(self) -> List[str]:
+        return []
+    
+    def validation(self) -> bool:
+        """config_validation connector setup"""
```
Comment (ddematheu) [inline neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py:27]: Any validation steps that can help make sure config is correct?

**Review by ddematheu — APPROVED** [review]

File: neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py (hunk 1 above)
Comment (sunilkumardash9) [reply neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py:27]: Would it be enough to initiate the inferenceClient with parameters?

**Comment by sunilkumardash9** [comment]
Hi @ddematheu, I was wondering if we should use the [lazy-imports](https://github.com/telekom/lazy-imports) library for deferred import of non-core dependencies like the transformer_hub.

File: neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py (hunk 1 above)
Comment (ddematheu) [reply neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py:27]: Yeah that would be great. We just want to ensure we can validate connections before we start processing data.

**Comment by ddematheu** [comment]
@sunilkumardash9 Haven't used lazy-imports before, but I think it makes sense for transformer_hub given the size of the import. Haven't really had the issue with other dependencies as they are more manageable.

**Comment by sunilkumardash9** [comment]
@ddematheu I see the Transformers library is already in the dependencies. So, it wouldn't make a difference here. But I think we can do it with task-specific dependencies like quadrant, Weaviate, and Marqo clients to reduce build dependencies.

**Comment by ddematheu** [comment]
You are right. Lets then just merge this PR and we can make the lazy import
changes separately.

**Comment by sunilkumardash9** [comment]
@ddematheu, Yes, I will make a separate PR for this.

**Pull request merged**